src/disambiguities.jl

```diff
@@ -1,14 +1,100 @@
+for (type, _, _) in ABSTRACT_QUANTITY_TYPES
+    @eval begin
+        Base.:(==)(::$type, ::WeakRef) = error("Cannot compare a quantity to a weakref")
+        Base.:(==)(::WeakRef, ::$type) = error("Cannot compare a weakref to a quantity")
+    end
+end
 
 Base.:*(l::AbstractDimensions, r::Number) = error("Please use an `UnionAbstractQuantity` for multiplication. You used multiplication on types: $(typeof(l)) and $(typeof(r)).")
 Base.:*(l::Number, r::AbstractDimensions) = error("Please use an `UnionAbstractQuantity` for multiplication. You used multiplication on types: $(typeof(l)) and $(typeof(r)).")
 Base.:/(l::AbstractDimensions, r::Number) = error("Please use an `UnionAbstractQuantity` for division. You used division on types: $(typeof(l)) and $(typeof(r)).")
 Base.:/(l::Number, r::AbstractDimensions) = error("Please use an `UnionAbstractQuantity` for division. You used division on types: $(typeof(l)) and $(typeof(r)).")
+
+# Promotion ambiguities
+function Base.promote_rule(::Type{F}, ::Type{Bool}) where {F<:FixedRational}
+    return F
+end
+function Base.promote_rule(::Type{Bool}, ::Type{F}) where {F<:FixedRational}
+    return F
+end
+function Base.promote_rule(::Type{F}, ::Type{BigFloat}) where {F<:FixedRational}
+    return promote_type(Rational{eltype(F)}, BigFloat)
+end
+function Base.promote_rule(::Type{BigFloat}, ::Type{F}) where {F<:FixedRational}
+    return promote_type(Rational{eltype(F)}, BigFloat)
+end
+function Base.promote_rule(::Type{F}, ::Type{T}) where {F<:FixedRational,T<:AbstractIrrational}
+    return promote_type(Rational{eltype(F)}, T)
+end
+function Base.promote_rule(::Type{T}, ::Type{F}) where {F<:FixedRational,T<:AbstractIrrational}
+    return promote_type(Rational{eltype(F)}, T)
+end
+
+################################################################################
+# Assorted calls found by Aqua: ################################################
+################################################################################
+
+for type in (Signed, Float64, Float32, Rational), op in (:flipsign, :copysign)
+    @eval function Base.$(op)(x::$type, y::AbstractRealQuantity)
+        return $(op)(x, ustrip(y))
+    end
+end
+function Base.:*(l::Complex, r::AbstractRealQuantity)
+    new_quantity(typeof(r), l * ustrip(r), dimension(r))
+end
+function Base.:*(l::AbstractRealQuantity, r::Complex)
+    new_quantity(typeof(l), ustrip(l) * r, dimension(l))
+end
+function Base.:/(l::Complex, r::AbstractRealQuantity)
+    new_quantity(typeof(r), l / ustrip(r), inv(dimension(r)))
+end
+function Base.:/(l::AbstractRealQuantity, r::Complex)
+    new_quantity(typeof(l), ustrip(l) / r, dimension(l))
+end
+function Base.:*(l::Complex{Bool}, r::AbstractRealQuantity)
+    return new_quantity(typeof(r), l * ustrip(r), dimension(r))
+end
+function Base.:*(l::AbstractRealQuantity, r::Complex{Bool})
+    return new_quantity(typeof(l), ustrip(l) * r, dimension(l))
+end
```

Comment:
This block of methods might be more readable as a metaprogramming loop? E.g. over the three pairs of (method, type)?